core/primitives/src/state.rs

```diff
@@ -102,4 +102,11 @@ impl FlatStateValue {
             Self::Inlined(value) => value.len(),
         }
     }
+
+    pub fn size(&self) -> usize {
```

Comment:
I'm using this new `size()` method because `value_len()` doesn't seem accurate to me, given the fact I'm literally copying only `FlatValues`